Linux cross-compilation broken since iconv cmake full TARGET commit

## Bug report
### Describe the bug

When building LibreELEC for RPi with current kodi HEAD df97ff523a99db66c9be6a779e79cbaa6caab7e5 linking kodi fails because it's trying to pull in /usr/lib/libc.so from the x86_64 build host (which is wrong as it should against the aarch64 libc from sysroot - full log is here: http://ix.io/4Iqh):

```
/home/hias/libreelec/libreelec-master/build.LibreELEC-RPi5.aarch64-12.0-devel/toolchain/lib/gcc/aarch64-libreelec-linux-gnu/13.2.0/../../../../aarch64-libreelec-linux-gnu/bin/ld.gold: error: cannot open /usr/lib/libc.so.6: No such file or directory
/home/hias/libreelec/libreelec-master/build.LibreELEC-RPi5.aarch64-12.0-devel/toolchain/lib/gcc/aarch64-libreelec-linux-gnu/13.2.0/../../../../aarch64-libreelec-linux-gnu/bin/ld.gold: error: cannot open /usr/lib/libc_nonshared.a: No such file or directory
/home/hias/libreelec/libreelec-master/build.LibreELEC-RPi5.aarch64-12.0-devel/toolchain/lib/gcc/aarch64-libreelec-linux-gnu/13.2.0/../../../../aarch64-libreelec-linux-gnu/bin/ld.gold: error: cannot open /usr/lib/ld-linux-aarch64.so.1: No such file or directory
...
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
FAILURE: scripts/build kodi:target during make_target (default)
```

Reverting the `[cmake] FindIconv migrate to full TARGET usage` commit b2c8bf3631a63bd41d4153aa245e186e26b2a0f6 fixes the build issue.

Analyzing the (verbose) ninja build and cmake trycompile steps showed that before the iconv cmake change the kodi link call used `-lc` whereas with the iconv change it uses `/home/hias/libreelec/libreelec-master/build.LibreELEC-RPi5.aarch64-12.0-devel/toolchain/aarch64-libreelec-linux-gnu/sysroot/usr/lib/libc.so` - which isn't a regular library but a ld script.

However cmake trycompile used `-lc` instead (which is somewhat OK)

CMakeFiles/CMakeScratch/TryCompile-qSFynL/build.ninja:
```
# Link the executable cmTC_187d0

build cmTC_187d0: C_EXECUTABLE_LINKER__cmTC_187d0_ CMakeFiles/cmTC_187d0.dir/CheckFunctionExists.c.o | /home/hias/libreelec/libreelec-master/build.LibreELEC-RPi5.aarch64-12.0-devel/toolchain/aarch64-libreelec-linux-gnu/sysroot/usr/lib/libc.so
  FLAGS = -march=armv8.2-a+crc+crypto -mabi=lp64 -Wno-psabi -mtune=cortex-a76 -mno-outline-atomics -Wall -pipe  -O3 -fomit-frame-pointer -DNDEBUG -Wdouble-promotion -Wmissing-field-initializers -Wsign-compare -Wextra -Wno-unused-parameter -DCHECK_FUNCTION_EXISTS=iconv
  LINK_FLAGS = -Wl,--as-needed -fuse-ld=gold -fuse-ld=gold
  LINK_LIBRARIES = -lc
  OBJECT_DIR = CMakeFiles/cmTC_187d0.dir
  POST_BUILD = :
  PRE_LINK = :
  TARGET_FILE = cmTC_187d0
  TARGET_PDB = cmTC_187d0.dbg
```

As `-lc` is unnecessary (libc is automatically linked in anyways) and linking against libc.so from sysroot will be wrong I'd guess the `c` library searching should just be dropped from the FindIconv.cmake script - iconv will be there, just like printf and anything else.

ping @fuzzard 

## Expected Behavior

Cross compiling kodi works as before

## Actual Behavior

Cross compiling kodi fails

## Possible Fix

### To Reproduce

Build LibreELEC master with kodi updated to current HEAD (see eg here: https://github.com/HiassofT/LibreELEC.tv/tree/le12-kodi-head)
```
PROJECT=RPi DEVICE=RPi5 ARCH=aarch64 make
```

### Debuglog

### Screenshots 


## Additional context or screenshots (if appropriate)

### Your Environment
Used Operating system:
<!--- Include as many relevant details about the environment you experienced the bug in -->
<!--- Put your text below this line. Checkboxes can easily be ticked once issue is created -->
 - [ ] Android
 - [ ] iOS
 - [ ] tvOS
 - [x] Linux
 - [ ] macOS
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name: LibreELEC master
 - Kodi version: Omega/master HEAD

--- a/cmake/modules/FindIconv.cmake
+++ b/cmake/modules/FindIconv.cmake
@@ -29,10 +29,14 @@ if(NOT TARGET ICONV::ICONV)
                                     REQUIRED_VARS ICONV_LIBRARY ICONV_INCLUDE_DIR HAVE_ICONV_FUNCTION)
 
   if(ICONV_FOUND)
-    add_library(ICONV::ICONV UNKNOWN IMPORTED)
-    set_target_properties(ICONV::ICONV PROPERTIES
-                                       IMPORTED_LOCATION "${ICONV_LIBRARY}"
-                                       INTERFACE_INCLUDE_DIRECTORIES "${ICONV_INCLUDE_DIR}")
-    set_property(GLOBAL APPEND PROPERTY INTERNAL_DEPS_PROP ICONV::ICONV)
+    # Libc causes grief for linux, so search if found library is libc.* and only
+    # create imported TARGET if its not
+    if(NOT ${ICONV_LIBRARY} MATCHES ".*libc\..*")
+      add_library(ICONV::ICONV UNKNOWN IMPORTED)
+      set_target_properties(ICONV::ICONV PROPERTIES
+                                         IMPORTED_LOCATION "${ICONV_LIBRARY}"
+                                         INTERFACE_INCLUDE_DIRECTORIES "${ICONV_INCLUDE_DIR}")
+      set_property(GLOBAL APPEND PROPERTY INTERNAL_DEPS_PROP ICONV::ICONV)
+    endif()
   endif()
 endif()

